boards/armv7m/default_handlers.h

```diff
@@ -131,24 +131,48 @@ void hard_fault_handler_step_3(void);
 __attribute__((__naked__)) static void hard_fault_handler(void)
 {
 // set switch to 0 in order to alternatively recreate the previous stack frame
-// and hald the CPU
+// and halt the CPU
 #if 1
     __asm volatile
     (
-        " tst   lr, #4               \n"
-        " ite   eq                   \n"
-        " mrseq r0, msp              \n"
-        " mrsne r0, psp              \n"
+        " mov r0, #4                 \n"
+        " mov r1, lr                 \n"
+        " tst r0, r1                 \n"
+
+#if 1 // code rewritten for ARMv6m
+        " bne has_bit_two            \n"
+        " mrs r0, msp                \n"
+        " b test_done                \n"
+        "has_bit_two:                \n"
+        " mrs r0, psp                \n"
+        "test_done:                  \n"
+#else // original code        
```

Comment:
Do we want to remove this dead code, or is it intentionally left behind?

Reply:
removed